### PR TITLE
GUARD-2952 (Access) Orders: Handle Optional and Deprecated Fields

### PR DESCRIPTION
**Pull request by maxim-skuvault**
[GUARD-2952]
Summary: Confirm that when optional eBay API orders fields aren't returned our code will still parse correctly. Don't parse deprecated fields and the fields we've identified as definitely not being saved into SkuVault. Fix the error when RefundAmount > "CurrencyCodeType" / “currencyID is missing.

#### Background
Numerous eBay API orders fields are optional. This can cause issues when they're not returned, and our response parsing code cannot handle them missing.

#### About these changes
For eBay GetOrdersResponseParser do:
- Create test for the case when optional ShippingAddress and ShippingService optional fields are missing
- Remove PaymentHoldDetails, ExpectedReleaseDate since they're not saved into SkuVault
- Add a test when MonetaryDetails.Refunds have missing fields
- Remove ShippingDetails.GetItFast deprecated field we don't pull. Update tests accordingly
- Remove ShippingPackageInfo since it's deprecated; add tests. Update tests accordingly
- Add tests of order.TransactionArray optional fields
- Remove order.RefundAmount, order.Refund.AmountCurrencyID since we weren't parsing it correctly and we don't sync it anyway. Update tests accordingly
- [Refactor] Move all GetOrdersResponseParser tests that verify behavior when optional fields are missing/empty to a separate class `EbayGetOrdersResponseParserOptionalFieldsTest.cs`

# PR Readiness Checklist

- [ ] I have updated all relevant configuration
- [x] Followed [development conventions][1] to the best of my ability
- [x] Code is documented, particularly public interfaces and hard-to-understand areas
- [x] Tests are updated / added and all pass
- [x] Performed a self-review of my own code
- [x] Acceptance criterias are confirmed by testing changes in UI (or another appropriate way)
- [ ] Confluence documentation is updated, if needed
- [x] New code does not generate new warnings

[1]: https://agileharbor.atlassian.net/wiki/spaces/DEV/pages/1114130/Conventions

[GUARD-2952]: https://agileharbor.atlassian.net/browse/GUARD-2952?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/EbayAccess/Services/Parsers/EbayGetOrdersResponseParser.cs [hunk 1]
```diff
@@ -189,10 +180,7 @@ public override GetOrdersResponse Parse( Stream stream, bool keepStreamPosition
 								var shippingServiceSelectedPackageInfo = shippingServiceSelected.Element( ns + "ShippingPackageInfo" );
 								if ( shippingServiceSelectedPackageInfo != null ) 
 								{
-									resTransaction.ShippingServiceSelected = new ShippingServiceSelected
-									{
-										ShippingPackageInfo = ParseShippingPackageInfo( shippingServiceSelectedPackageInfo, ns )
-									};
+									resTransaction.ShippingServiceSelected = new ShippingServiceSelected();
```
Comment (SShumer) [inline src/EbayAccess/Services/Parsers/EbayGetOrdersResponseParser.cs:183]: We don't use shippingServiceSelectedPackageInfo here, is that correct? Do we need the 
"new ShippingServiceSelected()" value?

File: src/Global/GlobalAssemblyInfo.cs [hunk 2]
```diff
@@ -24,4 +24,4 @@
 
 // Keep in track with CA API version
```
Comment (SShumer) [inline src/Global/GlobalAssemblyInfo.cs:25]: I saw it in different integrations. Does CA means Channeladvisor? Do we need it here?

Comment (maxim-skuvault) [reply src/Global/GlobalAssemblyInfo.cs:25]: A good catch. It does refer to ChannelAdvisor. Plus, we don't keep the package version "in track" (the same?) as the API version. So this comment isn't helpful. Will remove.

File: src/EbayAccess/Services/Parsers/EbayGetOrdersResponseParser.cs (hunk 1 above)
Comment (maxim-skuvault) [reply src/EbayAccess/Services/Parsers/EbayGetOrdersResponseParser.cs:183]: @SShumer I think we still need it. `ShippingServiceSelected` is also [returned in orders](https://github.com/skuvault-integrations/ebayAccess/pull/27/files/c95d248b6ce201f1d487fecdfa443c716150e625#diff-c44c14057778aa1bdec074667e4eb78d08dcd1d7f48b8d15c3220d2acb6ec8f6R103). There we do deserialize `ShippingService.ShippingServiceSelected`. Then in v1, we save it as shipping carrier: `order.ShippingServiceSelected?.ShippingService ?? order.ShippingDetails?.ShippingServiceOptions?.ShippingService`

**Review by SShumer — APPROVED** [review]

**Pull request merged**